src/gs-image-tile.c

```diff
@@ -0,0 +1,230 @@
+/* -*- Mode: C; tab-width: 8; indent-tabs-mode: t; c-basic-offset: 8 -*-
+ *
+ * Copyright (C) 2016 Endless Mobile, Inc.
+ * Copyright (C) 2013 Matthias Clasen <mclasen@redhat.com>
+ *
+ * Licensed under the GNU General Public License Version 2
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include "config.h"
+
+#include <glib/gi18n.h>
+#include <gtk/gtk.h>
+
+#include "gs-image-tile.h"
+#include "gs-star-widget.h"
+#include "gs-common.h"
+
+struct _GsImageTile
+{
+	GsAppTile	 parent_instance;
+
+	GsApp		*app;
+	GtkWidget	*app_name;
+	GtkWidget	*app_summary;
+	GtkWidget	*hover_app_name;
+	GtkWidget	*icon;
+	GtkWidget	*eventbox;
+	GtkWidget	*stack;
+	GtkWidget	*stars;
+	GtkWidget	*details_revealer;
+};
+
+G_DEFINE_TYPE (GsImageTile, gs_image_tile, GS_TYPE_APP_TILE)
+
+static GsApp *
+gs_image_tile_get_app (GsAppTile *tile)
+{
+	return GS_IMAGE_TILE (tile)->app;
+}
+
+static gboolean
+app_state_changed_idle (gpointer user_data)
+{
+	GsImageTile *tile = GS_IMAGE_TILE (user_data);
+	AtkObject *accessible;
+	GtkWidget *label;
+	gboolean installed;
+	g_autofree gchar *name = NULL;
+
+	accessible = gtk_widget_get_accessible (GTK_WIDGET (tile));
+
+	label = gtk_bin_get_child (GTK_BIN (tile->eventbox));
+	switch (gs_app_get_state (tile->app)) {
+	case AS_APP_STATE_INSTALLED:
+	case AS_APP_STATE_INSTALLING:
+	case AS_APP_STATE_REMOVING:
+	case AS_APP_STATE_UPDATABLE:
+	case AS_APP_STATE_UPDATABLE_LIVE:
+		installed = TRUE;
+		name = g_strdup_printf ("%s (%s)",
+					gs_app_get_name (tile->app),
+					_("Installed"));
```

Comment:
This does not seem particularly friendly to translators. What if the language does not use parenthesis like that? Perhaps the whole string `%s (Installed)` should be translatable, with a comment on what `%s` is.